sound/soc/sof/core.c

```diff
@@ -357,6 +357,14 @@ int snd_sof_device_probe(struct device *dev, struct snd_sof_pdata *plat_data)
 	sdev->first_boot = true;
 	dev_set_drvdata(dev, sdev);
 
+	/* check IPC support */
+	if (!(BIT(plat_data->ipc_type) & plat_data->desc->ipc_supported_mask)) {
+		dev_err(dev, "ipc_type %d is not supported on this platform, mask is %#x\n",
+			plat_data->ipc_type, plat_data->desc->ipc_supported_mask);
+		return -EINVAL;
```

Comment:
Will this really happen? We already test sof_pci_ipc_type on sof-pci-dev.c.

Reply:
Not strictly necessary but a good paranoia check. I wanted to double-check for the other cases (ACPI and OF) that we didn't miss a platform.